### PR TITLE
Add `curl` requirement for building paper.

### PR DESCRIPTION
**Pull request by katekerllenevich**
This adds the requirement of `curl` to CONTRIBUTING.md.

While this may seem obvious to many people, when I installed Debian on my machine, I wasn't able to build Paper and I wasn't able to figure it out until the next day when I needed to install curl for a separate program.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ you will most likely use this for WSL), `homebrew` (macOS / Linux), and more:
 If you need one, you can find them on [AdoptOpenJDK](https://adoptopenjdk.net/).
 - `maven` (often package `maven`; can be found on
 [Apache's site](https://maven.apache.org/download.cgi) too).
+- `curl` (package `curl` everywhere);
```
Comment (Proximyst) [inline CONTRIBUTING.md]:
```suggestion
[Apache's site](https://maven.apache.org/download.cgi) too);
- `curl` (package `curl` everywhere).
```

**Review by Proximyst — CHANGES_REQUESTED** [review]
Require it in `scripts/requireDeps.sh` too.

File: CONTRIBUTING.md (hunk 1 above)
Comment (katekerllenevich) [reply CONTRIBUTING.md]: haha i didnt know that was a thing, ill commit that in a second

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -63,7 +63,8 @@ you will most likely use this for WSL), `homebrew` (macOS / Linux), and more:
 - A Java 8 or later JDK (packages vary, use Google/DuckDuckGo/etc.).  
 If you need one, you can find them on [AdoptOpenJDK](https://adoptopenjdk.net/).
 - `maven` (often package `maven`; can be found on
-[Apache's site](https://maven.apache.org/download.cgi) too).
+[Apache's site](https://maven.apache.org/download.cgi) too);
```
Comment (Chew) [inline CONTRIBUTING.md:66]: Why the semicolon?

Comment (Chew) [reply CONTRIBUTING.md:66]: nvm, guess it's consistent with the ones above it

Comment (katekerllenevich) [reply CONTRIBUTING.md:66]: Git and Patch both have semi colons at the end, showing they aren't the end of the requirements. Java, for whatever reason, does not have a semi-colon.

**Review by Proximyst — APPROVED** [review]

**Review by zachbr — APPROVED** [review]
Thanks

**Pull request merged**